Rubberduck.VBEEditor/Class1.cs

```diff
@@ -0,0 +1,964 @@
+//
+// THIS IS A WORK IN PROGRESS.  ERROR HANDLING NEEDS WORK, AS DOES A FEW OF THE HELPER ROUTINES.
+//
+// WARNING: when using TypeLibsAccessor_VBE directly, do not cache it  
+//   The type library is LIVE information, so consider it a snapshot at that very moment when you are dealing with it
+//   Make sure you call TypeLibsAccessor_VBE.Dispose() as soon as you have done what you need to do with it.
+//   Once control returns back to the VBE, you must assume that all the ITypeLib/ITypeInfo pointers are now invalid.
+//
+// CURRENT LIMITATIONS:
+// At the moment, enums and UDTs are not exposed through the type libraries
+// Constants names are not available
+
+// IMPLEMENTATION DETAILS:
+// There are two significant bugs in the VBE typeinfos implementations that we have to work around.
+// 1)  Some implementations of ITypeInfo provided by the VBE will crash with an AV if you call IProvideClassInfo::GetClassInfo on them.
+//      And guess what method the CLR calls on all COM interop objects when creating a RCW?  You guessed it.
+//      So, we use an aggregation object, plus ITypeInfo and ITypeLib wrappers to circumvent this VBE bug.
+//
+// 2)  The ITypeInfo for base classes of UserForms crash with an AV if you call ITypeInfo::GetDocumentation(MEMBERID_NIL) to get the type name
+//     We've got to remember that the VBE didn't ever intend for us to get hold of these objects, so there will be little bugs.
+//     This bug is also resolved in the provided wrappers.
+//
+// All the extended functionality is exposed through the wrappers.
+
+namespace Rubberduck.VBEditor.TypeLibsAPI
+{
+    public class VBETypeLibsAPI
+    {
+        public static void ExecuteCode(IVBE ide, string projectName, string standardModuleName, string procName)
+        {
+            using (var typeLibs = new TypeLibsAccessor_VBE(ide))
+            {
+                typeLibs.FindTypeLib(projectName).FindTypeInfo(standardModuleName)
+                    .StdModExecute(procName, Reflection.BindingFlags.InvokeMethod);
+            }
+        }
+
+        public static string GetProjectConditionalCompilationArgs(IVBE ide, string projectName)
+        {
+            using (var typeLibs = new TypeLibsAccessor_VBE(ide))
+            {
+                return typeLibs.FindTypeLib(projectName).ConditionalCompilationArguments;
+            }
+        }
+
+        public static void SetProjectConditionalCompilationArgs(IVBE ide, string projectName, string newConditionalArgs)
+        {
+            using (var typeLibs = new TypeLibsAccessor_VBE(ide))
+            {
+                typeLibs.FindTypeLib(projectName).ConditionalCompilationArguments = newConditionalArgs;
+            }
+        }
+
+        public static bool IsAWorkbook(IVBE ide, string projectName, string className)
+        {
+            using (var typeLibs = new TypeLibsAccessor_VBE(ide))
+            {
+                return typeLibs.FindTypeLib(projectName).FindTypeInfo(className).DoesImplement("_Workbook");
```

Comment:
I provided an alternative implementation for DoesImplement() that takes the interface IID :)

Reply:
The IsAWorkbook/IsAWorksheet were just to demonstrate, but sure, we can make them more generic.

Reply:
Regarding implementation for IID - it might be useful to have a 3rd overload that can take a ProgID instead so we can do a lookup for IID and thus check IID without knowing the IID.